src/core/openthread-instance.h

```diff
@@ -100,6 +100,9 @@ typedef struct otInstance
     otLogLevel mLogLevel;
 #endif // OPENTHREAD_CONFIG_ENABLE_DYNAMIC_LOG_LEVEL
 
+    ot::Timer mTimer;
+    static void HandleInstanceTimer(void *aContext);
```

Comment:
This should be optional, as many (if not most) clients don't need this. I believe it should be optional and off/disabled by default.